Algorithms.Tests/Numeric/EulerMethodTest.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using Algorithms.Numeric;
+using NUnit.Framework;
+using System.Collections.Generic; 
+
+namespace Algorithms.Tests.Numeric
+{
+    public static class EulerMethodTest
+    {
+        [Test]
+        public static void TestLinearEquation()
+        {
+            Func<double, double, double> exampleEquation1 = (x, y) => x;
+            List<double[]> points =  Algorithms.Numeric.EulerMethod.EulerFull(0, 4, 0.1, 0, exampleEquation1);
+            double yEnd = points[points.Count - 1][1];
+            Assert.AreEqual(yEnd, 7.800000000000003);
+        }
+        
+        [Test]
+        public static void TestExampleWikipedia()
+        {
+            // example from https://en.wikipedia.org/wiki/Euler_method
+            Func<double, double, double> exampleEquation2 = (x, y) => y;
+            List<double[]> points =  Algorithms.Numeric.EulerMethod.EulerFull(0, 4, 0.1, 1, exampleEquation2);
+            double yEnd = points[points.Count - 1][1];
+            Assert.AreEqual(yEnd, 45.25925556817596);
+        }
+        
+        [Test]
+        public static void TestExampleGeeksForGeeks()
+        {
+            // example from https://www.geeksforgeeks.org/euler-method-solving-differential-equation/
+            Func<double, double, double> exampleEquation3 = (x, y) => x + y + x * y;
+            List<double[]> points =  Algorithms.Numeric.EulerMethod.EulerFull(0, 0.1, 0.025, 1, exampleEquation3);
+            double yEnd = points[points.Count - 1][1];
+            Assert.AreEqual(yEnd, 1.1116729841674804);

```

Comment:
Is this an analytical solution? We should only compare against it.

Reply:
The idea of this test was not to test the precision of the algorithm but to make sure that the implementation is correct. I changed it to focus only on the first 2 steps and added comments to make it more clear what is happening.

Reply:
Ok, I see. This is a good idea too!